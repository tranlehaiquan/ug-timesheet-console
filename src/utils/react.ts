import * as React from 'react'

type IdentityFunction = <T>(fn: T) => T

export const typedMemo: IdentityFunction = React.memo

export const getDisplayName = <T extends any = any>(WrappedComponent: React.ComponentType<T>): string => {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component'
}
