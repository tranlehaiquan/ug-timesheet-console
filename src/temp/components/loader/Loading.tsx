import * as React from 'react'
import classnames from 'classnames'

interface IProps {
  /**
   * className applied to the containing div
   */
  className?: string
  /**
   * Alignment of the component
   */
  align?: 'left' | 'center' | 'right'
}
/**
 * Loading indicator for lists
 */

export const Loading: React.FunctionComponent<IProps> = ({ className, align = 'center' }) => {
  const alignment = {
    left: 'sk-text-left',
    center: 'sk-text-center',
    right: 'sk-text-right'
  }

  return (
    <div className={ classnames('loading-throb', className, alignment[align]) }>
      <div className="loading-throb__bounce loading-throb__bounce--first" />
      <div className="loading-throb__bounce loading-throb__bounce--second" />
      <div className="loading-throb__bounce loading-throb__bounce--third" />
    </div>
  )
}
