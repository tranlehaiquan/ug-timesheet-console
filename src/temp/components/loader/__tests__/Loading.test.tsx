import * as React from "react";
import { render } from "enzyme";

import { Loading } from "../Loading";

describe("Loading", () => {
  test("center", () => {
    // Arrange
    const component = render(<Loading />);

    // Assert
    expect(component).toMatchSnapshot();
    expect(component.prop("class")).toContain("sk-text-center");
  });

  test("left", () => {
    // Arrange
    const component = render(<Loading align="left" />);

    // Assert
    expect(component).toMatchSnapshot();
    expect(component.prop("class")).toContain("sk-text-left");
  });

  test("right", () => {
    // Arrange
    const component = render(<Loading align="right" />);

    // Assert
    expect(component).toMatchSnapshot();
    expect(component.prop("class")).toContain("sk-text-right");
  });
});
