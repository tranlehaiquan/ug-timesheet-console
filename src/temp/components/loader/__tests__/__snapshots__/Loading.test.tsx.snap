// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading center 1`] = `
<div
  class="loading-throb sk-text-center"
>
  <div
    class="loading-throb__bounce loading-throb__bounce--first"
  />
  <div
    class="loading-throb__bounce loading-throb__bounce--second"
  />
  <div
    class="loading-throb__bounce loading-throb__bounce--third"
  />
</div>
`;

exports[`Loading left 1`] = `
<div
  class="loading-throb sk-text-left"
>
  <div
    class="loading-throb__bounce loading-throb__bounce--first"
  />
  <div
    class="loading-throb__bounce loading-throb__bounce--second"
  />
  <div
    class="loading-throb__bounce loading-throb__bounce--third"
  />
</div>
`;

exports[`Loading right 1`] = `
<div
  class="loading-throb sk-text-right"
>
  <div
    class="loading-throb__bounce loading-throb__bounce--first"
  />
  <div
    class="loading-throb__bounce loading-throb__bounce--second"
  />
  <div
    class="loading-throb__bounce loading-throb__bounce--third"
  />
</div>
`;
